### ISSUE TITLE
[$1000] Baseline of the recipient name and message preview do not line up in LHN

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14148]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open NewDot
2. Set `Priority mode` to focus in preferences
3. Observe the chat list in the LHN
 
## Expected Result:
Baseline of the recipient name and message preview should be aligned
![image](https://user-images.githubusercontent.com/14225673/211677975-d7264738-b4ef-4179-b2f9-3fbd75994260.png)

## Actual Result:
the baseline of the recipient name and message preview do not line up. 
![image](https://user-images.githubusercontent.com/14225673/211678017-06bf45ef-1709-4762-9307-129c0260f775.png)
Ideally the message preview line would be shifted downwards about 1-2px.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.50-14
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
<img width="375" alt="image (1)" src="https://user-images.githubusercontent.com/43996225/211379273-0a136681-d40a-4b73-a493-f625c50d2fc1.png">
<img width="1440" alt="Screen Shot 2023-01-09 at 1 14 07 PM" src="https://user-images.githubusercontent.com/43996225/211379467-2144a281-9bc2-44db-9fb1-3757abb8f8ed.png">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @shawnborton 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673202521146459

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e29f396b115e95bc</li>
        <li>Upwork Job ID: 1612943799700762624</li>
        <li>Last Price Increase: 2023-01-10</li>
    </ul>
</details>

**Comment by allroundexperts** [comment]
## Proposal

The issue is that the container for the content is using `align-items` as `center`. Instead, it should be `baseline`.

**Comment by aneequeahmad** [comment]
### Proposal

The `View` is having ` alignItems: 'center',` which is causing this issue
https://github.com/Expensify/App/blob/d5d89afb41c53c76f8df9e45d0d24b372588f460/src/components/LHNOptionsList/OptionRowLHN.js#L129-L134
### Fix

Changing it to alignItems: `flex-end` from `styles.alignItemsCenter,` will fix the issue.

https://user-images.githubusercontent.com/36359331/211684194-49ad677f-698f-426f-8f20-4b276a2abc55.mov

**Comment by Pujan92** [comment]
**Proposal**

Due to the font size variation the align difference exists. We can apply padding top 1px to align it.

```diff
diff --git a/src/styles/styles.js b/src/styles/styles.js
index 47ca20888..80493276e 100644
--- a/src/styles/styles.js
+++ b/src/styles/styles.js
@@ -1224,6 +1224,7 @@ const styles = {
         flexShrink: 1,
         flexGrow: 1,
         flexBasis: 'auto',
+        paddingTop: 1,
         ...optionAlternateTextPlatformStyles,
     },
```

**Comment by shawnborton** [comment]
Just a heads up that @aneequeahmad's proposal doesn't fix the issue. 

I do like how clean @allroundexperts's proposal is and it seems to do the trick, with the benefit of pushing the user name up just slightly for even more vertical alignment in that row overall. @chiragsalian this would be my recommendation but curious for your thoughts too.

**Comment by aneequeahmad** [comment]
### Proposal

**Problem:**

- `Text vertical center alignment issue` when there are two Texts in a line since recipient name has `fontSize 15px` and message preview has `fontSize 13px`. Also, `height` and `lineHeight` are '20px`.

**Solution**

**Case 1:** There is only one Text under View Ios: Use flex with alignItems/justifyContent to be more centered, Adr: height = lineHeight, includeFontPadding: false more centered

**Case 2:** There are two Texts in a line view and the two Text fonts are different, and the Text with a smaller font size does not need to use the height.

- The larger one is aligned using the above method. The smaller one can only be used with font-size, but not with the line-height and height attributes. In this case, both texts can be centered.

Case 3: There are two Texts in a line view and the two Text fonts are different, and the Text with a smaller font size must use the height.

- The one with the larger font size is aligned using the above method, and the smaller font size uses height with padding and includeFontPadding to achieve alignment.


In our case we have `Case 2` and we `don't need to use height and lineHeight` for message preview as it has smaller fontSize.

**Code changes will be** 

- Remove style `styles.optionAlternateText` which has height and lineHeight styles.

https://github.com/Expensify/App/blob/d5d89afb41c53c76f8df9e45d0d24b372588f460/src/components/LHNOptionsList/OptionRowLHN.js#L74-L75

 After applying the changes bug will be fixed as shown in video below.
 


https://user-images.githubusercontent.com/36359331/211939959-de89ed07-f311-416a-8a21-90e43c9285d7.mov

**Comment by aneequeahmad** [comment]
@Pujan92 i think you're right `align-items: center` and `justifyContent: 'center', //Centered horizontally`  is to align any content including text both horizontally and vertically. 

@allroundexperts i don't understand why the root cause is that `align-items: center` and it should be `baseline` ? Don't we want to center the content(text) both horizontally and vertically ?

**Comment by allroundexperts** [comment]
@anmurali I received an offer for this. Has someone from the C+ team reviewed this?

**Comment by JmillsExpensify** [comment]
@eVoloshchak @chiragsalian Do you mind weighting in on the proposals above, incorporating @shawnborton's [feedback](https://github.com/Expensify/App/issues/14148#issuecomment-1378348892)?

**Comment by Pujan92** [comment]
Agree with the @aneequeahmad on point that `align-items: baseLine` will consider the bottom of the text for all child items which leads to line height increment. 

<img width="376" alt="Screenshot 2023-01-12 at 12 00 34 PM" src="https://user-images.githubusercontent.com/14358475/211994217-367cc855-297a-4a68-b921-f0917c4791aa.png">

**New Proposal**

In our case, we don't require the same `lineheight` and `height` for the descriptive text. To make it align we can use the margin from the top after setting the content to the center(which is already set).

```diff
diff --git a/src/components/LHNOptionsList/OptionRowLHN.js b/src/components/LHNOptionsList/OptionRowLHN.js
index e64ed774b..ab2d34d6f 100644
--- a/src/components/LHNOptionsList/OptionRowLHN.js
+++ b/src/components/LHNOptionsList/OptionRowLHN.js
@@ -72,7 +72,7 @@ const OptionRowLHN = (props) => {
         ? [styles.ml1, StyleUtils.getBackgroundColorWithOpacityStyle(themeColors.icon, 0.5)]
         : [styles.ml1];
     const alternateTextStyle = StyleUtils.combineStyles(props.viewMode === CONST.OPTION_MODE.COMPACT
-        ? [textStyle, styles.optionAlternateText, styles.textLabelSupporting, styles.optionAlternateTextCompact, styles.ml2]
+        ? [textStyle, styles.textLabelSupporting, styles.optionAlternateTextCompact, styles.ml2]
         : [textStyle, styles.optionAlternateText, styles.textLabelSupporting], props.style);
     const contentContainerStyles = props.viewMode === CONST.OPTION_MODE.COMPACT
         ? [styles.flex1, styles.flexRow, styles.overflowHidden, styles.alignItemsCenter]
diff --git a/src/styles/styles.js b/src/styles/styles.js
index 3719a37e0..3bef38285 100644
--- a/src/styles/styles.js
+++ b/src/styles/styles.js
@@ -1219,6 +1219,7 @@ const styles = {
     },
 
     optionAlternateTextCompact: {
+        marginTop: 2,
         flexShrink: 1,
         flexGrow: 1,
         flexBasis: 'auto',
```

<img width="377" alt="Screenshot 2023-01-12 at 12 01 52 PM" src="https://user-images.githubusercontent.com/14358475/211994433-ec48adb9-5c5c-4553-9fa3-ac50ad6997ea.png">

**Comment by shawnborton** [comment]
> will consider the bottom of the text for all child items which leads to line height increment.

Personally I think this is fine, it doesn't actually increase the line height per se, just the height of the wrapper that wraps the sender and preview line. Either way though, it looks optically vertically centered which is what we want. So using `baseline` is still my preference.

**Comment by anmurali** [comment]
@allroundexperts - my bad that I jumped the process. Let's wait for @chiragsalian and @eVoloshchak to respond.

**Comment by eVoloshchak** [comment]
I think @allroundexperts's [proposal](https://github.com/Expensify/App/issues/14148#issuecomment-1378008022) is good, there are no additional margins and the change is quite simple. There are fair points about text container height increasing slightly, but this way we'll make sure the two items are **precisely** centered

🎀👀🎀 C+ reviewed!

**Comment by allroundexperts** [comment]
@eVoloshchak https://github.com/Expensify/App/pull/14257

**Comment by allroundexperts** [comment]
@shawnborton @eVoloshchak While I was creating the PR, it seems like `baseline` property does not work as expected on Android. 

https://github.com/facebook/react-native/issues/21918
<img width="338" alt="Screenshot 2023-01-13 at 5 11 51 AM" src="https://user-images.githubusercontent.com/30054992/212207576-b167b39b-4e5d-4e93-865a-4a54f7f46a39.png">

For Android, we can either:
a. keep the container as `align-items: center`.
b. add a padding / margin to the detail text as a fake to the baseline.

Let me know what you guys think.

**Comment by shawnborton** [comment]
Are you sure? It looks like a fix for this was merged: https://github.com/facebook/react-native/pull/31575

**Comment by allroundexperts** [comment]
@shawnborton Yes, but it's still an issue as reported by number of users.

**Comment by shawnborton** [comment]
Got it. I would prefer not to make a platform-specific exception so maybe we just go with the 1px padding approach instead.

**Comment by allroundexperts** [comment]
@shawnborton It turns out adding 1px padding does not work for all devices as well. On iOS 1px padding works fine but for web / desktop, its too less. I think the only solution here is to either use platform specific code or just do nothing.

**Comment by aneequeahmad** [comment]
@shawnborton You're right platform-specific solution isn't bullet proof, Also, adding 1px padding isn't bullet proof as well. 
This issue is fixed by [my proposal](https://github.com/Expensify/App/issues/14148#issuecomment-1379613652) which is simple and a well tested fact.
>  There are two Texts in a line view and the two Text fonts are different then the Text with a smaller font size can only be used with a different font-size when it doesn't the `line-height` and `height` attributes. In this case, both texts can be centered

I have tested on Android, IOS, Web/Desktop its working fine.

**Note:** 
Also i have noticed that the current issue also exist on Android but its less than 1px. Which is also fixed by my proposal.

**Comment by shawnborton** [comment]
Interesting @aneequeahmad - that seems to look better, but I would be curious how it impacts other parts of the app like the LHN when it's not in #focus mode.

**Comment by shawnborton** [comment]
As well as if that works across all platforms.

**Comment by Pujan92** [comment]
> 2. To make it align a bit down compare to username we can use the margin from the top after setting the content to the center(which is already set).
> 
> ```diff
> diff --git a/src/components/LHNOptionsList/OptionRowLHN.js b/src/components/LHNOptionsList/OptionRowLHN.js
> index e64ed774b..ab2d34d6f 100644
> --- a/src/components/LHNOptionsList/OptionRowLHN.js
> +++ b/src/components/LHNOptionsList/OptionRowLHN.js
> @@ -72,7 +72,7 @@ const OptionRowLHN = (props) => {
>          ? [styles.ml1, StyleUtils.getBackgroundColorWithOpacityStyle(themeColors.icon, 0.5)]
>          : [styles.ml1];
>      const alternateTextStyle = StyleUtils.combineStyles(props.viewMode === CONST.OPTION_MODE.COMPACT
> -        ? [textStyle, styles.optionAlternateText, styles.textLabelSupporting, styles.optionAlternateTextCompact, styles.ml2]
> +        ? [textStyle, styles.textLabelSupporting, styles.optionAlternateTextCompact, styles.ml2]
>          : [textStyle, styles.optionAlternateText, styles.textLabelSupporting], props.style);
>      const contentContainerStyles = props.viewMode === CONST.OPTION_MODE.COMPACT
>          ? [styles.flex1, styles.flexRow, styles.overflowHidden, styles.alignItemsCenter]
> diff --git a/src/styles/styles.js b/src/styles/styles.js
> index 3719a37e0..3bef38285 100644
> --- a/src/styles/styles.js
> +++ b/src/styles/styles.js
> @@ -1219,6 +1219,7 @@ const styles = {
>      },
>  
>      optionAlternateTextCompact: {
> +        marginTop: 2,
>          flexShrink: 1,
>          flexGrow: 1,
>          flexBasis: 'auto',
> ```

@shawnborton in case if we consider this proposal of margin top for descriptive text it will look like this.


<img width="385" alt="Screenshot 2023-01-16 at 7 00 41 PM" src="https://user-images.githubusercontent.com/14358475/212690391-12807b71-b96d-40eb-b28d-afc65b033c79.png">
<img width="330" alt="Screenshot 2023-01-16 at 7 01 27 PM" src="https://user-images.githubusercontent.com/14358475/212690401-06d9edac-dd6a-4a59-9b88-f4aa74145a5f.png">
<img width="349" alt="Screenshot 2023-01-16 at 7 02 18 PM" src="https://user-images.githubusercontent.com/14358475/212690406-a3a56661-d0ec-47f0-840c-3f5d363d1b8f.png">

**Comment by Pujan92** [comment]
@aneequeahmad actually @shawnborton wanted to descriptive text a bit down and not in the center and that's why I have added `marginTop` over [here(Point 2)](https://github.com/Expensify/App/issues/14148#issuecomment-1379874865) after taking out the class `optionAlternateText `.

**Comment by Pujan92** [comment]
I think 1px top margin then will make it on the same level with the name.

<img width="372" alt="Screenshot 2023-01-16 at 7 44 53 PM" src="https://user-images.githubusercontent.com/14358475/212698630-0a6d47c3-e304-48db-8069-b45bd2ca654d.png">

**Comment by shawnborton** [comment]
That looks better to me. Can you show me this same screenshot but with the sender name using Sentence case, like `Shawn Borton` or something as the sender? I wonder if actually we want to apply the 1px margin just to the bottom of the sender name to bump everything up just slightly? Hard to tell but a screenshot would help. Thanks!

**Comment by Pujan92** [comment]
<img width="378" alt="Screenshot 2023-01-16 at 7 53 04 PM" src="https://user-images.githubusercontent.com/14358475/212700343-1c32e70d-be88-4f57-aa35-106858eb2494.png">

**Comment by shawnborton** [comment]
Got it, I think that looks okay but I suppose it could all go up just by 1px to be more vertically centered.

**Comment by Pujan92** [comment]
In the mobile 1px seems to be less as it shows a bit high. 

**With 1px top margin**
<img width="718" alt="Screenshot 2023-01-16 at 7 59 30 PM" src="https://user-images.githubusercontent.com/14358475/212702079-f8229dbf-d497-429c-b485-85292112e437.png">

**With 2px top margin**
<img width="722" alt="Screenshot 2023-01-16 at 8 00 35 PM" src="https://user-images.githubusercontent.com/14358475/212702124-cc7242dd-1328-4f82-9a76-69d6e01148c6.png">

**Comment by Pujan92** [comment]
As we are setting the content to be center vertically, isn't it(user name) looking centered(or do you think we should move it a 1px up)?

<img width="378" alt="Screenshot 2023-01-16 at 8 08 14 PM" src="https://user-images.githubusercontent.com/14358475/212703786-1367f58c-b105-45b7-966c-63a12ce82ce3.png">

**Comment by shawnborton** [comment]
Can you fix your fonts on Android? Perhaps by pulling main, rebuilding, etc. But your Android screenshot is not using the right font. 

I am actually suggesting the opposite of your screenshots though, in that maybe we leave the message preview text where it is but move the sender name up by 1px, and we can do that by adding 1px to the bottom.

**Comment by Pujan92** [comment]
Ok, so for `#focus` after applying 2px bottom margin in the display name with lineHeight to 16 for alternate will look like this. For android I rebuild after latest pull but seeing the same font.

<img width="378" alt="Screenshot 2023-01-16 at 9 54 00 PM" src="https://user-images.githubusercontent.com/14358475/212725442-97c395c7-2b9e-4ebf-99bb-74216aab291b.png">

<img width="350" alt="Screenshot 2023-01-16 at 9 54 25 PM" src="https://user-images.githubusercontent.com/14358475/212725455-c86e47be-df09-48a7-95d9-2112f5abf346.png">

**Comment by shawnborton** [comment]
@Pujan92 can you please address my comments about the incorrect font on Android in your screenshots?

**Comment by shawnborton** [comment]
Hmm for the Android font issue, perhaps you can post in #expensify-open-source for more ideas there? It should be working on Android at least.

To clarify though, the first screenshot in your comment above is from web/mWeb or iOS with the correct font?

**Comment by aneequeahmad** [comment]
@shawnborton Sorry for stepping in late. 
> but I would be curious how it impacts other parts of the app like the LHN when it's not in #focus mode.
Below is the video attached on #focus mode and not in #focus.


https://user-images.githubusercontent.com/36359331/212732622-5f6ea154-cf9b-4b9d-bbec-0bf70793aa98.mov

> As well as if that works across all platforms.

I have tested on Android, IOS, Web/Desktop its [working fine ](https://github.com/Expensify/App/issues/14148#:~:text=I%20have%20tested%20on%20Android%2C%20IOS%2C%20Web/Desktop%20its%20working%20fine.)


Also @Pujan92 
> wanted to descriptive text a bit down and not in the center and that's why I have added marginTop over wafter taking out the class optionAlternateText .

- if you look at the video i attached with [my proposal](https://github.com/Expensify/App/issues/14148#:~:text=After%20applying%20the%20changes%20bug%20will%20be%20fixed%20as%20shown%20in%20video%20below.) you will see that the descriptive text moves a bit down not in center.

- `Adding a random hard coded value` like 1px for some specific platform and specific case to fix the UI is never recommended and not a good practice as it leads to more potential bugs. There shouldn't be platform checks to for the UI if there are any then there is something wrong with the logic and code.

- There is no root cause defined why 1px value is added what is the root cause that it fixes. ee

**Comment by shawnborton** [comment]
@aneequeahmad that looks good to me and seems better than doing the 1px hack. But re-reading your proposal, I am having a hard time following exactly what you are proposing to change. Are you saying that in #focus mode, you will essentially remove the lineHeight and height of the smaller text?

**Comment by Pujan92** [comment]
> To clarify though, the first screenshot in your comment above is from web/mWeb or iOS with the correct font?
It is on the web.

@aneequeahmad Can you plz share Screencast of android once?

**Comment by allroundexperts** [comment]
With @aneequeahmad changes:
<img width="370" alt="Screenshot 2023-01-17 at 4 11 04 AM" src="https://user-images.githubusercontent.com/30054992/212778443-9bf06017-1e6a-466c-83d8-3d289ed5c32d.png">

Witout @aneequeahmad changes:
<img width="375" alt="Screenshot 2023-01-17 at 4 11 20 AM" src="https://user-images.githubusercontent.com/30054992/212778446-9171fcf2-163e-4d85-a54e-59f11fe8de27.png">

**Comment by allroundexperts** [comment]
cc @shawnborton It seems like android is still off by a pixel or so. We can either close this without any action or release a fix for other platforms at-least.

**Comment by shawnborton** [comment]
Really? It looks like Android is fixed based on your comment [here](https://github.com/Expensify/App/issues/14148#issuecomment-1384663163). 

@chiragsalian curious for your thoughts but I think we should go with @aneequeahmad 's proposal.

**Comment by aneequeahmad** [comment]
> Are you saying that in #focus mode, you will essentially remove the lineHeight and height of the smaller text?

Yes, that's correct and removing `lineHeight and height` and for both modes i-e `#focus and no focus` works fine. Waiting for your final call @shawnborton. Thanks

**Comment by shawnborton** [comment]
I think we only need to adjust this for #focus mode, otherwise I think your proposal is good. Waiting for @chiragsalian to confirm too.

**Comment by aneequeahmad** [comment]
@shawnborton Yeah that makes sense. We only need this fix for `#focus` mode. 
Let's wait for the confirmation.

**Comment by chiragsalian** [comment]
Sorry i'm not sure how i missed this earlier. Catching up on this now. So for starters i did like @allroundexperts first [proposal](https://github.com/Expensify/App/issues/14148#issuecomment-1378008022) since its simple clean and feels straightforward. Its a bummer it does not work well on android.

Okay next I'm looking at @aneequeahmad proposal [here](https://github.com/Expensify/App/issues/14148#issuecomment-1379613652). I like the proposal but when testing on web locally I see it off by like maybe 1px. (notice the user email is touching/on the line while their preview message is hovering 1 or 2px above)
![Screenshot 2023-01-19 at 2 38 53 PM](https://user-images.githubusercontent.com/14356145/213578417-2bd0d5f4-60c4-4dc9-85e1-f2b4bf614ac4.png) 

Is my understanding of the changes correct i.e., from the proposal you are suggesting the following,
```diff
diff --git a/src/components/LHNOptionsList/OptionRowLHN.js b/src/components/LHNOptionsList/OptionRowLHN.js
index e64ed774b8..ab2d34d6f0 100644
--- a/src/components/LHNOptionsList/OptionRowLHN.js
+++ b/src/components/LHNOptionsList/OptionRowLHN.js
@@ -72,7 +72,7 @@ const OptionRowLHN = (props) => {
         ? [styles.ml1, StyleUtils.getBackgroundColorWithOpacityStyle(themeColors.icon, 0.5)]
         : [styles.ml1];
     const alternateTextStyle = StyleUtils.combineStyles(props.viewMode === CONST.OPTION_MODE.COMPACT
-        ? [textStyle, styles.optionAlternateText, styles.textLabelSupporting, styles.optionAlternateTextCompact, styles.ml2]
+        ? [textStyle, styles.textLabelSupporting, styles.optionAlternateTextCompact, styles.ml2]
         : [textStyle, styles.optionAlternateText, styles.textLabelSupporting], props.style);
     const contentContainerStyles = props.viewMode === CONST.OPTION_MODE.COMPACT
         ? [styles.flex1, styles.flexRow, styles.overflowHidden, styles.alignItemsCenter]
```

**Comment by allroundexperts** [comment]
@chiragsalian There are too many edge cases here. I still feel like `text-align: baseline` is the way to go here. Its the standard way to make sure that all texts align vertically. This seems like a bug in android. Why can't we just add a android specific style?

**Comment by aneequeahmad** [comment]
@chiragsalian Yes, these right [these](https://github.com/Expensify/App/issues/14148#:~:text=you%20are%20suggesting%20the%20following) are then changes that i'm suggesting.

 I don't see any gap on web i just tested again on macOs(chrome)

Screenshots:
![Screen Shot 2023-01-20 at 4 11 31 AM](https://user-images.githubusercontent.com/36359331/213583078-5928b5f8-1813-4513-9164-c56ccea9d9d7.png)


![Screen Shot 2023-01-20 at 4 15 30 AM](https://user-images.githubusercontent.com/36359331/213583338-192a1337-335e-42dd-bd2e-acddf9470b18.png)

**Comment by chiragsalian** [comment]
@allroundexperts, I'm personally just not a fan of platform-specific styles. @shawnborton is our CSS guru though so if it's okay with him then I'm fine with it 😂 I feel like I'm more fine keeping it slightly misaligned on android than having a platform-specific style which feels a bit hacky. Because once it's fixed in react-native it should hopefully be fixed for android users.

@aneequeahmad, hmm that's odd. I also tested on web on macOS chrome and i see the message off by 1 or 2px 🤷‍♂️ Maybe @eVoloshchak can double-check.

**Comment by aneequeahmad** [comment]
Also my [proposal](https://github.com/Expensify/App/issues/14148#issuecomment-1379613652) fixes the other issue of text alignment raised [here](https://github.com/Expensify/App/issues/14269). Which was off on Android and the [general rule](https://github.com/Expensify/App/issues/14148#:~:text=Case%201%3A,to%20achieve%20alignment.) to align two text in a same line is a valid and well tested solution for all platforms and we don't need any platform specific check.

What i have learned from working on startup apps from scratch and building pixel perfect apps is that

- It isn't a good practice to give height to every UI element. Height value should only be used for containers. The content inside that container if coded right using CSS the height is perfect and matches exactly the design.

- Platform checks shouldn't be applied to fix the UI related issues and there should be single code that is written for cross platform apps and `react-native` is there already to take care of these platform things.

**Comment by eVoloshchak** [comment]
> @aneequeahmad, hmm that's odd. I also tested on web on macOS chrome and i see the message off by 1 or 2px 🤷‍♂️ Maybe @eVoloshchak can double-check.

<img width="1025" alt="image" src="https://user-images.githubusercontent.com/9059945/213586912-88b71c9c-bccf-4696-935c-2a229e44dd86.png">

@chiragsalian, I also see the message being off by 2px

I'm confused as to why first message is more off that the second
<img width="1123" alt="Screenshot 2023-01-20 at 01 39 39" src="https://user-images.githubusercontent.com/9059945/213587119-670fec74-c808-410e-9ba9-64b8223d17e3.png">

**Comment by aneequeahmad** [comment]
@eVoloshchak Hmmm that's strange i can see both `recipient name and message preview` are perfectly aligned even i did last comment as `test9`

![Screen Shot 2023-01-20 at 5 43 25 AM](https://user-images.githubusercontent.com/36359331/213594057-40de0b00-6843-4493-a8f7-b4f812986bdf.png)

**Comment by chiragsalian** [comment]
@aneequeahmad can you make sure your code is up to date before making those changes?

Also wait a min, in your own screenshot they don't look aligned. Are you sure you are making the same code changes vs live CSS edits?
![Screenshot 2023-01-19 at 5 13 37 PM](https://user-images.githubusercontent.com/14356145/213597060-0e480d0a-b4fb-42d1-a935-9facce661326.png)

**Comment by aneequeahmad** [comment]
@chiragsalian Yeah i just checked again for the confirmation and my code is latest. `v1.2.57-0`

The last screenshot i posted were the live CSS changes. Apologies for the confusion. 
Im attaching a screen-recording of the whole process. Please verify.

**Screen Recording:**

https://user-images.githubusercontent.com/36359331/213602055-8e38f252-0523-47fa-8768-900a16b73dec.mov

**Screen shot:**


![Screen Shot 2023-01-20 at 6 59 02 AM](https://user-images.githubusercontent.com/36359331/213602118-aa512a11-9476-4480-a164-59fb9ebb570d.png)

**Comment by eVoloshchak** [comment]
I think the issue we're having is that Chrome and Safari are behaving differently on macOS. 
This is the same page with @aneequeahmad's proposal applied
<img width="1507" alt="image" src="https://user-images.githubusercontent.com/9059945/214039877-347eba4a-8a2a-4ba8-8356-5f0be9814b5f.png">

**Comment by aneequeahmad** [comment]
@eVoloshchak i can see perfect alignment by applying my proposal on both `Chrome and Safari`.

**Chrome:** 

![Screen Shot 2023-01-24 at 1 52 42 AM](https://user-images.githubusercontent.com/36359331/214147611-a8836988-f573-4716-98d5-cab96aeb62e7.png)

**Comment by eVoloshchak** [comment]
> @eVoloshchak i can see perfect alignment by applying my proposal on both `Chrome and Safari`.

I think I might be applying your fix wrong, could you please share a branch?

**Comment by aneequeahmad** [comment]
@eVoloshchak Please checkout this branch and have a look

**Branch:**
 https://github.com/aneequeahmad/App/tree/fix-message-preview-alignment

**Comment by eVoloshchak** [comment]
@aneequeahmad, they are still misaligned on my screen by 2px it seems
<img width="692" alt="image" src="https://user-images.githubusercontent.com/9059945/214647791-fb4bea7b-b3b5-410b-bd04-46f577202add.png">

And pinned chat with Concierge is more misaligned (but this could be due to icon and/or emoji)

<img width="364" alt="image" src="https://user-images.githubusercontent.com/9059945/214647985-c90554b1-5e4a-4969-a93e-579a7e61fb89.png">

**Comment by chiragsalian** [comment]
@aneequeahmad, yeah it's also misalinged on my screen

@shawnborton, any suggestions here? What are your thoughts if we go ahead with `text-align: baseline` and add an android specific style?

**Comment by shawnborton** [comment]
I'd be cool with that - I know we try to avoid platform-specific code but at the same time we're doing this because we're trying to achieve the same exact style everywhere.

**Comment by aneequeahmad** [comment]
@chiragsalian, @eVoloshchak I dont understand they are fixed and perfectly aligned at my end.

**Comment by aneequeahmad** [comment]
@eVoloshchak, @chiragsalian Could you please share some more details about your setup and testing environment(macOS(year, intel/M1 chip, version of macOs, Chrome, Safari, version of Chrome, Safari,  ? So i could reproduce this scenario ? 

I can see perfect alignment at my end on all platforms. I d

**Comment by aneequeahmad** [comment]
@chiragsalian  This what you suggest [here](https://github.com/Expensify/App/issues/14148#:~:text=What%20are%20your%20thoughts%20if%20we%20go%20ahead%20with%20text%2Dalign%3A%20baseline%20and%20add%20an%20android%20specific%20style%3F) is a hack not a bullet proof solution in IMO. We can maybe get more people to test this branch maybe

**Comment by chiragsalian** [comment]
@aneequeahmad, yeah i'm not sure how we're seeing different things. This is what i did,
1. Cloned your branch, then i ran `npm i && npm run web`.
2. When i logged into my account the info is slightly misaligned.

![image](https://user-images.githubusercontent.com/14356145/214687617-c523e663-0d95-4e23-ad3e-f7c41fa48733.png)

MacOS - ventura 13.1
2021 MacBook pro, 14inch M1 chip.
I'm testing on Chrome - version 109.0.5414.87

Can you also share a screenshot where we can see multiple rows of users and their preview message.

**Comment by eVoloshchak** [comment]
Same specs and Chrome version as @chiragsalian, the only difference is Ventura 13.0. Also tested in Safari 16.1 (18614.2.9.1.12)

**Comment by eVoloshchak** [comment]
It seems like applying `alignItems: 'baseline'` will give us pixel-perfect alignment only on web and desktop. iOS it's also not aligned, tested both @aneequeahmad' and @allroundexperts's proposals. 
react-native has support for [alignItems: 'baseline'](https://reactnative.dev/docs/flexbox#align-items), I think it should be possible to align two text views

**Comment by allroundexperts** [comment]
@eVoloshchak What do you see for iOS? Can you post a screenshot?

**Comment by eVoloshchak** [comment]
![Simulator Screen Shot - iPhone 12 - 2023-01-25 at 23 36 42](https://user-images.githubusercontent.com/9059945/214696881-eeb68b4b-5f65-4e2c-aa38-201c1e78151f.png)

**Comment by aneequeahmad** [comment]
@eVoloshchak, @chiragsalian, @shawnborton Finally i figured out why my proposal for the fix is not consistent. 

**Problem:**

Actually my proposal fixes the alignment issue as long as there is no emoji in the last message(Since adding emoji increase the lineHeight). 

**Consider these 2 test cases.**

- Send last message without any emoji. (Pixel perfect alignment can be seen)

- Send last message with emoji. (message preview do not line up due to line height increase because of emoji). Edit this message and remove emoji from the last message(Pixel perfect alignment can be seen)

Screen recording attached:


https://user-images.githubusercontent.com/36359331/214715280-dc9c428c-5159-4448-9318-33ae09757e79.mov

**Comment by aneequeahmad** [comment]
### Solution for the inconsistent behavior
`(due to adding emoji in last message lineHeight increases from 15 to 21)`.

If we add `lineHeight: 15` in 

  ```diff
    optionAlternateTextCompact: {
         flexShrink: 1,
         flexGrow: 1,
         flexBasis: 'auto',
  +     lineHeight: 15,
         ...optionAlternateTextPlatformStyles,
    },

```

it fixes the issue and i can't see any regression on any platform.

Screenshot:

https://user-images.githubusercontent.com/36359331/215331471-44f39f89-9033-4047-8b0d-90326d63660a.mov

cc: @shawnborton @eVoloshchak

**Comment by shawnborton** [comment]
Could we use something like 20px or 16px for our lineHeight? That way it matches existing line heights we use in the app.

**Comment by eVoloshchak** [comment]
> ### Solution for the inconsistent behavior
> `(due to adding emoji in last message lineHeight increases from 15 to 21)`.

@aneequeahmad, weird, it is still misaligned on web on my screen

<img width="1474" alt="Screenshot 2023-01-30 at 23 04 19" src="https://user-images.githubusercontent.com/9059945/215594824-4a4aced0-0c05-4af5-bef1-89f77f352b6c.png">
<img width="1474" alt="Screenshot 2023-01-30 at 23 03 57" src="https://user-images.githubusercontent.com/9059945/215594834-77f5df7d-fb18-412c-ae8b-c4651ae37c7f.png">

I think we need to address the root cause, which is `alignItems: 'baseline'` not working for Android. And there might be another issue with iOS, since it is also slightly misaligned even though `alignItems: 'baseline'`  works there

**Comment by eVoloshchak** [comment]
> Could we use something like 20px or 16px for our lineHeight? That way it matches existing line heights we use in the app.

I'll test this tomorrow, but just looking at the code we'd need to use `lineHeightXLarge`

**Comment by aneequeahmad** [comment]
@eVoloshchak Actually the root cause has been fixed with my proposal on the other issue where name and date was misaligned. 

The issue here is that when there is an emoji it increases lineHeight of the message preview due to presence of emoji.

**Comment by allroundexperts** [comment]
@eVoloshchak Shouldn't this be bumped?

**Comment by eVoloshchak** [comment]
> Could we use something like 20px or 16px for our lineHeight? That way it matches existing line heights we use in the app.

We're using `lineHeightXLarge` (20-24) for both text items, they should match

**Comment by eVoloshchak** [comment]
> @eVoloshchak Actually the root cause has been fixed with my proposal on the other issue where name and date was misaligned.
> 
> The issue here is that when there is an emoji it increases lineHeight of the message preview due to presence of emoji.

It's still misaligned on my screen, take a look at the screenshots [here](https://github.com/Expensify/App/issues/14148#issuecomment-1409349300)

**Comment by eVoloshchak** [comment]
> @eVoloshchak Shouldn't this be bumped?

Thank you, it should!

It seems like we still don't have a proposal that would resolve the issue on all platforms, so awaiting proposals

**Comment by allroundexperts** [comment]
@eVoloshchak I spent some more time on this over the weekend. In my opinion, we should fix this at least on web and desktop (where we're sure about achieving a pixel perfect result). We can add android/iOS specific designs to make it all aligned. It seems like there is no silver bullet that will fix this on all platforms since we're using wrapped texts and pills.

**Comment by aneequeahmad** [comment]
@eVoloshchak My proposal is used to fix the other misalignment [issue](https://github.com/Expensify/App/issues/14269). There is still a minor misalignment on that issue but @shawnborton said [here](https://github.com/Expensify/App/issues/14269) this can be ignored as it doesn't feel broken.

So we can do the same here and using my proposal the text is misaligned only for the edge case when there is emoji present in the last message. Also, we don't need to add any platform check(it isn't a good practice and is discouraged by the community)

@shawnborton using `line-height 16px` make it a little misaligned only for the case when there is an emoji present in the last message.

**Comment by eVoloshchak** [comment]
@aneequeahmad, yeah, I think your proposal is the closest we can get to being pixel-perfect. It's still misaligned on my screen by 1px, but you have to zoom in to notice, so I agree this doesn't feel broken.

I'm not that strongly against platform-specific files if they can provide a better solution, so I think we could fix `web/mWeb/Desktop` with @allroundexperts's solution of `alignItemsBaseline` and native platforms with @aneequeahmad's solution. 
@shawnborton, what do you think?

**Comment by shawnborton** [comment]
I would be curious for @chiragsalian's opinion there since that's more of a technical question.

**Comment by chiragsalian** [comment]
I'm not sure what the exact question is. Is the question, 
1. if I am against platform-specific files? If so then no i don't mind it when its needed.
2. What is our preference? I think if we could create a reusable single style that also has a platform-specific style but can be reused in future for similar style issues I think that would be great.

I like the idea of `alignItemsBaseline` for everything except native platforms and then have something specific for native platforms.

**Comment by eh2077** [comment]
**Proposal**



##### Please re-state the problem that we are trying to solve in this issue.

If we edit a sent header bold text, `# heading` , and save it in chat, there'll be an extra leading line break above the header.

##### What is the root cause of that problem?

When translating html `h1` tag to markdown, we simply wrap it with a leading and a trailing line break. For example, html 

```html
<h1>heading</h1>
```

will be translated into markdown string

`\n# heading\n`

If we change the markdown text to

`\n# heading test\n`

and save it, then markdown text

`\n# heading test\n`

will be translated to html

```html
<h1><br />heading test<h1>
```

by applying markdown-to-html rules

https://github.com/Expensify/expensify-common/blob/668be7533215fc8d683d658da839361664320965/lib/ExpensiMark.js#L164-L173

The leading line break will be translated to the extra `<br />` tag.

##### What changes do you think we should make in order to solve the problem?

To fix this issue, we should improve the regex rule to translate `h1` tag to markdown. Instead of simply wrapping it with a leading and trailing line break, we should only add line break if there's non-empty text before or after the `<h1>` heading. We can improve the regex rule from 

https://github.com/Expensify/expensify-common/blob/668be7533215fc8d683d658da839361664320965/lib/ExpensiMark.js#L232-L233

to the following

```javascript
regex: /([\s\S]*?)\s*<(h1)(?:"[^"]*"|'[^']*'|[^'">])*>(.*?)<\/\2>(?![^<]*(<\/pre>|<\/code>))\s*([\s\S]*)/gi,
replacement: (match, g1, g2, g3, g4, g5) => `${g1}${g1 && '\n'}# ${g3}${g5 && '\n'}${g5}`,
```

So the html

```html
<h1>heading</h1>
```

will be translated to markdown text

`# heading`

and the edited markdown text

`# heading test` will be translated into html

```html
<h1>heading test</h1>
```

##### What alternative solutions did you explore? (Optional)

Call trim method for markdown text

**Comment by eh2077** [comment]
**Proposal**



### Please re-state the problem that we are trying to solve in this issue.

If we edit a sent header bold text, `# heading` , and save it in chat, there'll be an extra leading line break above the header.

### What is the root cause of that problem?

When translating html `h1` tag to markdown, we simply wrap it with a leading and a trailing line break. For example, html 

```html
<h1>heading</h1>
```

will be translated into markdown string

`\n# heading\n`

If we change the markdown text to

`\n# heading test\n`

and save it, then markdown text

`\n# heading test\n`

will be translated to html

```html
<h1><br />heading test<h1>
```

by applying markdown-to-html rules

https://github.com/Expensify/expensify-common/blob/668be7533215fc8d683d658da839361664320965/lib/ExpensiMark.js#L164-L173

The leading line break will be translated to the extra `<br />` tag.

### What changes do you think we should make in order to solve the problem?

To fix this issue, we should improve the regex rule to translate `h1` tag to markdown. Instead of simply wrapping it with a leading and trailing line break, we should only add line break if there's non-empty text before or after the `<h1>` heading. We can improve the regex rule from 

https://github.com/Expensify/expensify-common/blob/668be7533215fc8d683d658da839361664320965/lib/ExpensiMark.js#L232-L233

to the following

```javascript
regex: /([\s\S]*?)\s*<(h1)(?:"[^"]*"|'[^']*'|[^'">])*>(.*?)<\/\2>(?![^<]*(<\/pre>|<\/code>))\s*([\s\S]*)/gi,
replacement: (match, g1, g2, g3, g4, g5) => `${g1}${g1 && '\n'}# ${g3}${g5 && '\n'}${g5}`,
```

So the html

```html
<h1>heading</h1>
```

will be translated to markdown text

`# heading`

and the edited markdown text

`# heading test` will be translated into html

```html
<h1>heading test</h1>
```

### What alternative solutions did you explore? (Optional)

Call trim method for markdown text

**Comment by eh2077** [comment]
**Proposal**



### Please re-state the problem that we are trying to solve in this issue.

If we edit a sent header bold text, `# heading` , and save it in chat, there'll be an extra leading line break above the header.

### What is the root cause of that problem?

When translating html `h1` tag to markdown, we simply wrap it with a leading and a trailing line break. For example, html 

```html
<h1>heading</h1>
```

will be translated into markdown string

`\n# heading\n`

If we change the markdown text to

`\n# heading test\n`

and save it, then markdown text

`\n# heading test\n`

will be translated to html

```html
<h1><br />heading test<h1>
```

by applying markdown-to-html rules

https://github.com/Expensify/expensify-common/blob/668be7533215fc8d683d658da839361664320965/lib/ExpensiMark.js#L164-L173

The leading line break will be translated to the extra `<br />` tag.

### What changes do you think we should make in order to solve the problem?

To fix this issue, we should improve the regex rule to translate `h1` tag to markdown. Instead of simply wrapping it with a leading and trailing line break, we should only add line break if there's non-empty text before or after the `<h1>` heading. We can improve the regex rule from 

https://github.com/Expensify/expensify-common/blob/668be7533215fc8d683d658da839361664320965/lib/ExpensiMark.js#L232-L233

to the following

```javascript
regex: /([\s\S]*?)\s*<(h1)(?:"[^"]*"|'[^']*'|[^'">])*>(.*?)<\/\2>(?![^<]*(<\/pre>|<\/code>))\s*([\s\S]*)/gi,
replacement: (match, g1, g2, g3, g4, g5) => `${g1}${g1 && '\n'}# ${g3}${g5 && '\n'}${g5}`,
```

So the html

```html
<h1>heading</h1>
```

will be translated to markdown text

`# heading`

and the edited markdown text

`# heading test`

will be translated into html

```html
<h1>heading test</h1>
```

### What alternative solutions did you explore? (Optional)

Call trim method for markdown text

**Comment by aneequeahmad** [comment]
@chiragsalian There was another same misalignment [issue](https://github.com/Expensify/App/issues/14269) of name and date and that is fixed and deployed on production. 

There is no android specific check used in this fix, here is the [PR](https://github.com/Expensify/App/pull/14304) for code reference. Maybe @luacmartins can give an opinion that would help us to decide the solution for this problem and also [create a reusable single style](https://github.com/Expensify/App/issues/14148#:~:text=create%20a%20reusable%20single%20style%20that%20also%20has%20a%20platform%2Dspecific%20style%20but%20can%20be%20reused%20in%20future%20for%20similar%20style%20issues)

**Comment by allroundexperts** [comment]
@luacmartins we require your input here.

**Comment by luacmartins** [comment]
@allroundexperts sorry, this is quite a long discussion. What exactly do you need my input on?

**Comment by aneequeahmad** [comment]
@luacmartins this issue is identical as the other misalignment [issue](https://github.com/Expensify/App/issues/14269) we fixed. We have 2 proposals here. Need to decide consistent and bullet proof solution. Also, need to make the code consistent to be used for similar issues

### Proposal 1

> The issue is that the container for the content is using align-items as center. Instead, it should be baseline.

- This [Proposal](https://github.com/Expensify/App/issues/14148#issuecomment-1378008022) selected at first but it makes the android platform more broken as in screenshot below and to fix this case platform(android) specific check was proposed to be added by the proposal giver. 

![Screen Shot 2023-02-14 at 5 17 35 AM](https://user-images.githubusercontent.com/36359331/218606094-8c76dc1f-c7bc-42f1-a500-50446f7afa0c.jpg)

### Proposal 2

- This is my [proposal](https://github.com/Expensify/App/issues/14148#issuecomment-1379613652) it is the same we used to fix the other name and timestamp misalignment issue. It fixed the issue on all platforms but there is a minor misalignment for one specific edge case on web only when there is an emoji in the last message as adding emoji in last message increases the line height. [but you have to zoom in to notice, so I agree this doesn't feel broken.](https://github.com/Expensify/App/issues/14148#issuecomment-1419030861) 

Here is the screenshot of the misalignment.

![Screen Shot 2023-02-14 at 5 51 35 AM](https://user-images.githubusercontent.com/36359331/218609875-d40b263a-dc80-4977-8f5b-7597ed2e894d.png)


This is the situation here. IMO platform specific check isn't a bullet-proof solution and we already used concept/code from my proposal to fix the other misalignment issue and applying the same code should be used here to fix the issue which will also make the code consistent.

What's your opinion @luacmartins ?

**Comment by allroundexperts** [comment]
@luacmartins On web, there couldn't be a better fix then using the native `align-items: baseline`. Since its not implemented properly on react-native, that does not mean that we should not use it on web. I think combining both of these solutions make more sense here.

**Comment by aneequeahmad** [comment]
@luacmartins Yes, `Proposal 2` is the one we implemented. It works perfectly fine here. There is one edge case when there is an emoji in the last message it increases the line-height and then the `recipient name and message preview` is misaligned by 1px, but you have to zoom in to notice, so this doesn't feel broken as said by @eVoloshchak [here](https://github.com/Expensify/App/issues/14148#issuecomment-1419030861).

Root cause here and on the other issue is one of the cases that i proposed and works in every case and is a proper fix that can be used for future misalignment issues when there are two Texts in a line and have different font sizes.

**Comment by chiragsalian** [comment]
Yeah personally I'm a fan of combining both the solutions as stated here,
> I like the idea of alignItemsBaseline for everything except native platforms and then have something specific for native platforms.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by aneequeahmad** [comment]
@mvtglobally The issue is still reproducible on production. Please look at the video below.


https://user-images.githubusercontent.com/36359331/219069765-e98f1992-26a0-4426-831c-ac814f3903c9.mov

**Comment by allroundexperts** [comment]
@eVoloshchak @chiragsalian @aneequeahmad Can we reach a conclusion here please? This has been pending for some time now.

**Comment by aneequeahmad** [comment]
@allroundexperts I think majority agrees in combining both the solutions for pixel perfection. Let's do it and close this issue. 

cc: @chiragsalian @shawnborton @eVoloshchak

**Comment by shawnborton** [comment]
Works for me. Thoughts @chiragsalian?

**Comment by chiragsalian** [comment]
Yup, combining both solutions works for me too. I'll give @eVoloshchak a chance to review too.

**Comment by eVoloshchak** [comment]
I agree, combining both solutions is the best approach
🎀👀🎀 C+ reviewed!

**Comment by chiragsalian** [comment]
Sounds great. Assigning @aneequeahmad to complete the work.

**Comment by MelvinBot** [comment]
📣 @aneequeahmad You have been assigned to this job by @chiragsalian!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by allroundexperts** [comment]
@aneequeahmad If you have limited availability, I can create a PR as well.

**Comment by aneequeahmad** [comment]
@chiragsalian sounds good, I'll create the PR today around 9 pm PST. Thanks

@allroundexperts if you are available before this time you can create the PR as well. Thanks for your kind comment.

**Comment by allroundexperts** [comment]
@aneequeahmad I'm on it.

**Comment by allroundexperts** [comment]
PR created:
https://github.com/Expensify/App/pull/15319

**Comment by allroundexperts** [comment]
@aneequeahmad Feel free to make any changes if required.

**Comment by allroundexperts** [comment]
@aneequeahmad There's an issue on Android with alignment. Can you take a look?
https://github.com/Expensify/App/pull/15319

**Comment by MelvinBot** [comment]
@shawnborton, @eVoloshchak, @anmurali, @chiragsalian, @aneequeahmad Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@shawnborton, @eVoloshchak, @anmurali, @chiragsalian, @aneequeahmad Eep! 4 days overdue now. Issues have feelings too...

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.77-4 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15375


If no regressions arise, payment will be issued on 2023-03-09. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@eVoloshchak / @chiragsalian] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@eVoloshchak / @chiragsalian] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@eVoloshchak / @chiragsalian] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@aneequeahmad] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@anmurali] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by eVoloshchak** [comment]
This issue was linked to the PR by a mistake, PR for this issue wasn't merged yet

**Comment by MelvinBot** [comment]
@shawnborton, @eVoloshchak, @anmurali, @chiragsalian, @aneequeahmad Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@shawnborton, @eVoloshchak, @anmurali, @chiragsalian, @aneequeahmad Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.88-2 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15319


If no regressions arise, payment will be issued on 2023-03-31. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@eVoloshchak / @chiragsalian] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@eVoloshchak / @chiragsalian] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@eVoloshchak / @chiragsalian] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@anmurali] Determine if we should create a regression test for this bug.
- [ ] [@eVoloshchak] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@anmurali] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by eVoloshchak** [comment]
Not overdue, all that's left is to finish the BugZero Checklist, will do the first 3 items tomorrow

**Comment by aneequeahmad** [comment]
Any payment updates here ? Seems like this issue has been solved in production quite a while ago.

**Comment by chiragsalian** [comment]
@anmurali, any update here wrt payment?

**Comment by aneequeahmad** [comment]
@anmurali waiting for the update

**Comment by aneequeahmad** [comment]
This issue is 15 days late since payment date. I think @anmurali might be OOO.

cc: @shawnborton, @chiragsalian, @eVoloshchak

**Comment by anmurali** [comment]
Sorry for delay, everyone's been invited to a contract and/or paid now.

**Comment by anmurali** [comment]
@eVoloshchak @chiragsalian do we need to add a regression test for this?

**Comment by aneequeahmad** [comment]
@anmurali Thankyou for inviting. accepted the offer.

**Comment by eVoloshchak** [comment]
**Regression Test Proposal**

1. Set Priority mode to `focus` in preferences
2. Observe the chat list in the LHN
3. Verify each of the items is displayed correctly in the LHN, baseline of the recipient name and message preview should be aligned

Do we agree 👍 or 👎

**Comment by shawnborton** [comment]
Not overdue, but what's the latest on this one?

**Comment by eVoloshchak** [comment]
@anmurali, could you take a look at [this](https://github.com/Expensify/App/issues/14148#issuecomment-1515440822) please?

**Comment by shawnborton** [comment]
Not overdue

**Comment by anmurali** [comment]
I don't see a dupe payment @allroundexperts but Upwork is so confusing, so I will take your word for it. Requested a refund.

**Comment by anmurali** [comment]
I think we're all done here and can close this.
